tests/helpers/main.ts

```diff
@@ -49,19 +70,21 @@ export const zipNode = async function (
 
 export const getBundlerNameFromOptions = ({ config = {} }: { config?: Config }) =>
   config['*'] && config['*'].nodeBundler
+
 export const zipFixture = async function (
   fixture: string[] | string,
   { length, fixtureDir, opts = {} }: ZipOptions = {},
 ): Promise<ZipReturn> {
   const bundlerString = getBundlerNameFromOptions(opts) || 'default'
   const { path: tmpDir } = await getTmpDir({
     prefix: `zip-it-test-bundler-${bundlerString}`,
-    // Cleanup the folder on process exit even if there are still files in them
-    unsafeCleanup: true,
+    keep: ZISI_KEEP_TEMP_DIRS,
   })
 
-  if (env.ZISI_KEEP_TEMP_DIRS !== undefined) {
+  if (ZISI_KEEP_TEMP_DIRS) {
     console.log(tmpDir)
+  } else {
+    cleanupDirectories.push(tmpDir)
```

Comment:
Instead of removing all temporary directories that start with `zip-it-test-bundler-`, we now collect all directories and remove them after all tests.